torch/csrc/jit/codegen/cuda/kernel_cache.h

```diff
@@ -21,16 +23,63 @@ namespace cuda {
 // TODO: Figure out how we want to cache based on heuristics, should probably
 // use something similar. Heuristics may also return a LaunchParams object.
 // TODO: Validate it is included in the string.
+
 class FusionExecutorCache {
  public:
-  FusionExecutor* getExecutor() const {
-    return entry;
-  }
+  // FusionExecutorCache(Fusion* fusion, CompileOptions options);
+  FusionExecutorCache(std::unique_ptr<Fusion>&& fusion, at::Device device);
+
+  std::vector<at::Tensor> runFusionWithInputs(
+      const at::ArrayRef<IValue>& inputs);
+
+ private:
+  at::Device device_;
+  std::unique_ptr<Fusion> fusion_;
+  std::vector<std::unique_ptr<FusionExecutor>> fusion_executor_cache_;
```

Comment:
Perhaps this is preliminary, but I figured maybe a hash_map of fusions?

Reply:
Sorry for the confusion, it is just a place holder here.
In the Heuristic PR I have unordered_map for reduction cache: https://github.com/csarofeen/pytorch/pull/275/files#diff-5cf1af9e7456fd87042cea5d7adfd888R44